### ISSUE TITLE
GET /ldap/groups/search API endpoint does not return the full UserGroup object

### ISSUE THREAD

**Issue by sdamico918** [goharbor/harbor#15171]
**Expected behavior:**
When querying the `GET /ldap/groups/search` API endpoint using the `groupname` and `groupdn` parameters, an output of type `UserGroup` is returned:
```
[
  {
    "group_name": "string",
    "ldap_group_dn": "string",
    "group_type": 0,
    "id": 0
  }
]
```
This is what is reported in the Swagger API as well: 
![image](https://user-images.githubusercontent.com/72998396/122848223-f706c500-d2bd-11eb-9be9-3fadf971dc78.png)

**Actual behavior:**
Querying the `GET /ldap/groups/search` endpoint with parameters `groupname=mygroup` only returns the `group_name` and `group_dn`:
```
[
  {
    "group_name": "mygroup",
    "ldap_group_dn": "ou=Groups,ou=myenv,ou=Environments,dc=mydomain,dc=io"
  }
]
```
The `id` and `group_type` fields are missing. This makes it impossible to fetch a group's ID, as the `GET /usergroups` endpoint does not allow filtering by individual groups.

**Steps to reproduce the problem:**
On any harbor instance, go to the Swagger API page and navigate to `GET /ldap/groups/search`, then execute a sample command with a valid `groupname` parameter. A valid curl command will get the same results.

**Versions:**

- harbor version: [2.1.4]
- docker engine version: [20.10.4]
- docker-compose version: [1.21.2]

**Comment by stonezdj** [comment]
This API is only for search ldap group by groupName or LDAP group DN, it doesn't return the group ID because it might not created in the Harbor and the group type is ldap group. we are going to remove theses redundant attributes.

**Comment by sdamico918** [comment]
> This API is only for search ldap group by groupName or LDAP group DN, it doesn't return the group ID because it might not created in the Harbor and the group type is ldap group. we are going to remove these redundant attributes.

If a group isn't created, what difference does it make whether the return object contains just the name/DN or name/DN/ID? Also, the reason I am trying to fetch the ID is because I have to add an LDAP group member to multiple projects, and the documentation under the `POST /projects/{project_id}/members` endpoint says:
```
The group_member need to specify id or ldap_group_dn. If the group already exist in harbor DB. specify the user group's id, If does not exist, it will SearchAndOnBoard the group.
```
If the group already exists, then I would need to fetch the group's ID in order to add it as a member to more projects. Adding the LDAP group via DN was causing issues if the group already existed in the Harbor database - the group was not showing up as a member in the UI despite me getting a 201 from the request that was supposed to add it. I was hoping to resolve this by adding an LDAP group by its ID, but it seems there is no way I can do that if this endpoint is not meant to return the full `UserGroup` object.

**Comment by sdamico918** [comment]
I wanted to provide some more context on my use-case:

I have an LDAP group that I want to add to multiple projects. Up until this point, my LDAP groups have been 1-1 with projects, so I've been using the `POST /projects/{project_id}/members` endpoint to import the LDAP groups via DN _and_ add them to the project all at once. Now that I have a group that needs access to multiple projects, I tried reusing my code that hits the `POST /projects/{project_id}/members` endpoint, but it fails with a 409 because the LDAP group has already been imported to Harbor _even though it is not a member of the project I am trying to add it to_. Thus, there seems to be no way to add an already-existent LDAP group to a project via the LDAP group's DN. This is why I am trying to get the LDAP group's ID, which I _can_ use to add an already-existent LDAP group to a project (and this is in alignment with description of the `POST /projects/{project_id}/members` endpoint, which I copied into my response above).

If the `GET /ldap/groups/search` endpoint is not meant to return the group ID, how else can I get the ID? To reiterate, the `GET /usergroups` endpoint is not sufficient because that endpoint does not allow filtering of the responses. Theoretically, I could loop through all the responses, determine if the `group_name` in each indexed response matches what I'm looking for, and then use that index to fetch the ID, but that is impractical and unnecessary. The ID is clearly available to the API and, according to the docs, should be returned by `GET /ldap/groups/search` anyways.

My ask is either that the `GET /ldap/groups/search` endpoint is modified to return the entire `UserGroup` object as is specified in the documentation, or an additional endpoint is added from which the ID of a previously-imported LDAP group could be obtained.
